modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/NavigationMenuResourceImpl.java

```diff
@@ -580,15 +684,40 @@ private NavigationMenuItem _toNavigationMenuItem(
 							"getSiteSitePage", contextUriInfo,
 							arguments.toArray(new Object[0]));
 					});
-				setUseCustomName(
+
+				setContentURL(
 					() -> {
-						if (layout == null) {
+						if ((layout != null) ||
+							StringUtil.equals(type, "url") ||
+							type.contains("asset") ||
+							StringUtil.equals(type, "navigationMenu")) {
+
 							return null;
 						}
 
-						return Boolean.valueOf(
-							unicodeProperties.getProperty(
-								"useCustomName", "false"));
+						if (Objects.equals(type, FileEntry.class.getName())) {
+							type = DLFileEntry.class.getName();
+						}
+
+						Long classPK = GetterUtil.getLong(
+							unicodeProperties.getProperty("classPK"));
+
+						List<Object> arguments = new ArrayList<>();
+
+						String applicationPath = _getApplicationPath(
+							_getApplicationName(type, classPK));
+
+						Class<?> clazz = _getClassNameContent(type, classPK);
+
+						String methodName = _getMethodName(type, classPK);
+
+						Long contentId = _getItemId(unicodeProperties);
+
+						arguments.add(contentId);
+
+						return JaxRsLinkUtil.getJaxRsLink(
+							applicationPath, clazz, methodName, contextUriInfo,
```

Comment:
I saw that you are now using the applicationPath but still not generating URLs for entities outside `headless-delivery`. I supposed that you would do the task of supporting the other entities in the future, right?